static/report/js/compare.js

```diff
@@ -0,0 +1,637 @@
+      .call(make_y_axis( )
+        .tickSize(-width , 0, 0)
+        .tickValues([40, 80, 120,  180, 240, 300,  400])
+        .tickFormat("")
+        .orient("left")
+        // .ticks(6)
+      )
+
+  }
+  function render_circles (dots) {
+    dots
+        .transition( )
+        .attr("r", 5)
+        .attr("title",  function (d) { return d.key; })
+        .attr("fill", function (d) { return d.values.color; })
+        .attr("cx", function (d) { return scales.x(d.values.target); })
+        .attr("cy", function (d) { return scales.y(d.values.mean); })
+    return dots;
+  }
+
+  function render (selection) {
+    console.log('selection');
+  }
+
+  my.setRange = function _setRange (begin, end) {
+    adjuster(begin, end);
+  }
+
+  function adjust_range (begin, end) {
+    my.page.refresh(begin, end);
+  }
+
+  var adjuster = _.debounce(adjust_range, 1000);
+
+  function on_data (payload) {
+    var days = time_in_range(payload, opts);
+    my.data = days;
+    my( );
+    // console.log('got days', days);
+  }
+
+  function init ( ) {
+    my.page = pager({begin: opts.begin, end: opts.end, callback: on_data, cache: opts.cache});
+    get_dimensions( );
+
+    scales.x = d3.scale.linear( )
+      .domain([0, 100])
+      .range([0, width])
+      ;
+
+    scales.y = d3.scale.log( )
+          .domain( [40, 400] )
+          .rangeRound( [height - (margin.top + margin.bottom), 1] )
+          .base(12)
+      ;
+
+
+    // Axis
+    xAxis = make_x_axis( )
+      .ticks(10)
+      .tickFormat(function(d) { return parseInt(d, 10) + "%"; })
+      .orient('bottom')
+      ;
+
+
+    yAxis = make_y_axis( )
+      .tickValues([40, 60, 70, 80, 120, 160, 180, 200, 220, 260, 300, 350, 400])
+      .tickFormat(d3.format("d"))
+      .tickSize(6, 3, 1)
+      .orient('left')
+      ;
+
+    root = d3.select(dom.get(0)).append('svg')
+      .attr('class', 'ranger')
+      .attr('height', dom_height)
+      .attr('width', dom_width)
+      ;
+
+    chart = root.append('g')
+      .attr("transform", "translate(" + 0 + ", " + (margin.top) + ")")
+      .attr('class', 'ranger-chart')
+      ;
+
+    chart.append('g').attr('class', 'x gridlines')
+      .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      ;
+    chart.append('g').attr('class', 'y gridlines')
+      .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      ;
+    dots = chart.append("g")
+      .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      .attr("class", "scatter")
+      ;
+    chart.append("g")
+      .attr("transform", "translate(" + margin.left + ", " + (height - margin.top) + ")")
+      .attr("class", "x axis")
+      .call(xAxis)
+      ;
+    chart.append("g")
+      .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      // .attr("transform", "translate(" + dom_width + ", " + (0) + ")")
+      .attr("class", "y axis")
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2165148169/issues/source?bid=2452379&fileBranchId=2967311#l590)